syntax = "proto3";

package lvs;

import "google/protobuf/struct.proto";

service LVSService {
  rpc BulkLVS (BulkLVSRequest) returns (BulkLVSResponse);
  rpc LayoutGen (LayoutGenRequest) returns (SiVistaResponse);
  rpc GTSRemap (RemapRequest) returns (SiVistaResponse);
  rpc Version (Empty) returns (SiVistaResponse);
}

message BulkLVSRequest {
  string cell_name = 1;
  string netlist_path = 2;
  float scaling_factor = 3;
  int32 width = 4;
  int32 length = 5;
  string technology = 6;
  string layer_stack = 7;
  string gdsjsons = 8;
  int32 height_req = 9;
}

message BulkLVSResponse {
  map<string, string> result = 1;
}

message LayoutGenRequest {
  string args = 1;
}

message SiVistaResponse {
  string status = 1;
  string message = 2;
}

message RemapRequest {
  string technology = 1;
  string input_folder = 2;
  string output_folder = 3;
  map<string, google.protobuf.Value> params = 4;
}

message Empty {}